pkg/volume/rbd/attacher.go

```diff
@@ -214,6 +215,18 @@ func (detacher *rbdDetacher) UnmountDevice(deviceMountPath string) error {
 	}
 	klog.V(3).Infof("rbd: successfully umount device mountpath %s", deviceMountPath)
 
+	if devicePath == "" {
+		deviceMountedPathSeps := strings.Split(deviceMountPath, "-image-")
+		if len(deviceMountedPathSeps) != 2 {
+			return fmt.Errorf("Can't found devicePath for %v", deviceMountPath)
+		}
+		found := false
+		devicePath, found = getRbdDevFromImageAndPool(deviceMountedPathSeps[0], deviceMountedPathSeps[1])
+		if !found {
+			return fmt.Errorf("Can't found devicePath for %v", deviceMountPath)
```

Comment:
If not found, log a warning log only and pass. Because what we need to do here is to unmount device, if we could not find it in our best, then we can think it was unmounted already (by previous action or other parities, e.g sys admins). We must go on. Otherwise, we may enter into infinite loop.